Revit_Core_Engine/Modify/Checkout.cs

```diff
@@ -0,0 +1,85 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Modifies CheckoutStatus for each element selected if element is not currently owned by the current user or others.")]
+        [Input("elements", "Revit elements to query for its checkout status.")]
+        [Output("elementsCheckedOut", "List of elements checked out (ownership modified to be by current user) by this process.")]
+        public static List<Element> Checkout(this List<Element> elements)
+        {
+            if (elements == null || elements.Count <= 0)
+            {
+                BH.Engine.Base.Compute.RecordError("Element list cannot be null or empty.");
+                return null;
+            }
+
+            var elementsOfSameDocument = elements.Select(x => x.Document);
+
+            if (elementsOfSameDocument.Distinct().Count() > 1)
+            {
+                BH.Engine.Base.Compute.RecordError("Elements cannot be from different Revit documents.");
+                return null;
+            }
+
+            Document document = elementsOfSameDocument.First();
+            
+            List<Element> elementsToCheckout = new List<Element>();
+
+            foreach (Element element in elements)
+            {
+                switch (element.CheckoutStatus())
+                {
+                    case CheckoutStatus.NotOwned:
+                        elementsToCheckout.Add(element);
+                        break;
+                    case CheckoutStatus.OwnedByCurrentUser:
+                        Compute.ElementOwnedByCurrentUserNote(element);
+                        break;
+                    case CheckoutStatus.OwnedByOtherUser:
+                        Compute.ElementOwnedByOtherUserWarning(element);
+                        break;
+                }
+            }
+
+            WorksharingUtils.CheckoutElements(document, elementsToCheckout.Select(x => x.Id).ToList());
```

Comment:
This could be done as follows:
```suggestion
            foreach (var group in elementsToCheckout.GroupBy(x => x.Document))
            {
                WorksharingUtils.CheckoutElements(group.Key, group.Select(x => x.Id).ToList());
            }
```